saleor/checkout/calculations.py

```diff
@@ -117,9 +117,27 @@ def calculate_checkout_total_with_gift_cards(
         lines=lines,
         address=address,
         database_connection_name=database_connection_name,
-    ) - checkout_info.checkout.get_total_gift_cards_balance(database_connection_name)
+    )
+
+    gift_cards_balance = checkout_info.checkout.get_total_gift_cards_balance(
+        database_connection_name
+    )
+    if gift_cards_balance.amount > 0:
+        total_with_gift_cards = max(
+            total - gift_cards_balance, zero_taxed_money(total.currency)
+        )
+        gift_card_compensation = min(total.gross, gift_cards_balance)
+        details = {
+            "checkout_id": str(checkout_info.checkout.pk),
+            "gift_card_compensation": str(gift_card_compensation.amount),
+            "total_after_gift_card_compensation": str(
+                total_with_gift_cards.gross.amount
+            ),
+        }
+        logger.info("Gift card payment.", extra=details)
```

Comment:
I'm not sure if this is the right place, as we just have a calculation here. And it will be logged each time prices for checkout are recalculated. Also, the many from gift card is not reduced here. Maybe `add_gift_cards_to_order` is a better place?

Reply:
make sense, thanks